flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/adapter/DefaultSchedulingPipelinedRegionTest.java

```diff
@@ -0,0 +1,127 @@
+		final DefaultExecutionVertex vertex = new DefaultExecutionVertex(
+			new ExecutionVertexID(new JobVertexID(), 0),
+			Collections.emptyList(),
+			() -> ExecutionState.CREATED,
+			InputDependencyConstraint.ANY);
+
+		final Set<DefaultExecutionVertex> vertices = Collections.singleton(vertex);
+		final DefaultSchedulingPipelinedRegion pipelinedRegion = new DefaultSchedulingPipelinedRegion(vertices);
+		final Iterator<DefaultExecutionVertex> vertexIterator = pipelinedRegion.getVertices().iterator();
+
+		assertThat(vertexIterator.hasNext(), is(true));
+		assertThat(vertexIterator.next(), is(sameInstance(vertex)));
+		assertThat(vertexIterator.hasNext(), is(false));
+	}
+
+	/**
+	 * Tests if the consumed inputs of the pipelined regions are computed
+	 * correctly using the Job graph below.
+	 * <pre>
+	 *          c
+	 *        /  X
+	 * a -+- b   e
+	 *       \  /
+	 *        d
+	 * </pre>
+	 * Pipelined regions: {a}, {b, c, d, e}
+	 */
+	@Test
+	public void returnsIncidentBlockingPartitions() throws Exception {
+		final JobVertex a = ExecutionGraphTestUtils.createNoOpVertex(1);
+		final JobVertex b = ExecutionGraphTestUtils.createNoOpVertex(1);
+		final JobVertex c = ExecutionGraphTestUtils.createNoOpVertex(1);
+		final JobVertex d = ExecutionGraphTestUtils.createNoOpVertex(1);
+		final JobVertex e = ExecutionGraphTestUtils.createNoOpVertex(1);
+
+		b.connectNewDataSetAsInput(a, DistributionPattern.POINTWISE, ResultPartitionType.BLOCKING);
+		c.connectNewDataSetAsInput(b, DistributionPattern.POINTWISE, ResultPartitionType.PIPELINED);
+		d.connectNewDataSetAsInput(b, DistributionPattern.POINTWISE, ResultPartitionType.PIPELINED);
+		e.connectNewDataSetAsInput(c, DistributionPattern.POINTWISE, ResultPartitionType.BLOCKING);
+		e.connectNewDataSetAsInput(d, DistributionPattern.POINTWISE, ResultPartitionType.PIPELINED);
+
+		final ExecutionGraph simpleTestGraph = ExecutionGraphTestUtils.createSimpleTestGraph(a, b, c, d, e);
+		final DefaultExecutionTopology topology = new DefaultExecutionTopology(simpleTestGraph);
+
+		final DefaultSchedulingPipelinedRegion firstPipelinedRegion = topology.getPipelinedRegionOfVertex(new ExecutionVertexID(a.getID(), 0));
+		final DefaultSchedulingPipelinedRegion secondPipelinedRegion = topology.getPipelinedRegionOfVertex(new ExecutionVertexID(e.getID(), 0));
+
+		final DefaultExecutionVertex vertexB0 = topology.getVertex(new ExecutionVertexID(b.getID(), 0));
+		final IntermediateResultPartitionID b0ConsumedResultPartition = Iterables.getOnlyElement(vertexB0.getConsumedResults()).getId();
+
+		final Set<IntermediateResultPartitionID> secondPipelinedRegionConsumedResults = IterableUtils.toStream(secondPipelinedRegion.getConsumedResults())
+			.map(DefaultResultPartition::getId)
+			.collect(Collectors.toSet());
+
+		assertThat(firstPipelinedRegion.getConsumedResults().iterator().hasNext(), is(false));
+		assertThat(secondPipelinedRegionConsumedResults, contains(b0ConsumedResultPartition));
```

Comment:
maybe assert equals to verify that the result partition produced by c0 is not a consumed partition of the second region?

Reply:
It's already implied:
> [...] For a positive match, the examined iterable must be of the same length as the number of specified items.

Reply:
Ah, my bad. Always misunderstand this `contains`.